packages/web-wallet/src/components/WalletHome.tsx

```diff
@@ -0,0 +1,312 @@
+                </p>
+              )}
+            </div>
+
+            {/* Right: Action Buttons */}
+            <div className="flex gap-3 w-full md:w-auto">
+              <button
+                onClick={() => handleButtonClick('Send')}
+                className={`flex-1 md:flex-none px-6 py-2.5 bg-primary hover:bg-primary/90 rounded-xl flex items-center justify-center gap-2 transition-colors ${activeButton === 'Send' ? 'bg-primary/90' : ''
+                  }`}
+              >
+                <ArrowUpRight className="w-4 h-4 text-white" />
+                <span className="text-sm font-medium text-white">Send</span>
+              </button>
+              <button
+                onClick={() => handleButtonClick('Receive')}
+                className={`flex-1 md:flex-none px-6 py-2.5 bg-primary hover:bg-primary/90 rounded-xl flex items-center justify-center gap-2 transition-colors ${activeButton === 'Receive' ? 'bg-primary/90' : ''
+                  }`}
+              >
+                <ArrowDownLeft className="w-4 h-4 text-white" />
+                <span className="text-sm font-medium text-white">Receive</span>
+              </button>
+            </div>
+          </div>
+        </div>
+
+        {/* My Assets Section */}
+        <div className="space-y-6">
+          {/* Section Header */}
+          <h2 className="text-xl font-medium text-white">My Assets</h2>
+
+          {/* Token Tabs */}
+          <TokenTabs
+            selectedFilter={selectedTokenFilter}
+            onFilterChange={(filter) => updateSearchParams({ filter })}
+            tokenCount={tokenCount}
+            nftCount={nftCount}
+          />
+
+          {/* Token List */}
+          <TokenList
+            tokens={tokens}
+            onTokenClick={(tokenUid) => {
+              navigate(`?dialog=history&token=${tokenUid}`);
+            }}
+            onSendClick={(tokenUid) => {
+              navigate(`?dialog=send&token=${tokenUid}`);
+            }}
+          />
+        </div>
+      </div>
+
+      {/* Dialogs */}
+      <SendDialog
+        isOpen={sendDialogOpen}
+        onClose={closeDialog}
+        initialTokenUid={tokenUid || undefined}
+      />
+      <ReceiveDialog
+        isOpen={receiveDialogOpen}
+        onClose={closeDialog}
+      />
+      <HistoryDialog
+        isOpen={historyDialogOpen}
+        onClose={closeDialog}
+        tokenUid={tokenUid || undefined}
+        onRegisterTokenClick={() => navigate('?dialog=register')}
+      />
+      <RegisterTokenDialog
+        isOpen={registerTokenOpen}
+        onClose={closeDialog}
+      />
+      <CreateTokenDialog
+        isOpen={createTokenOpen}
+        onClose={closeDialog}
+      />
+      <AddressModeDialog
+        isOpen={addressModeOpen}
+        onClose={closeDialog}
+      />
+
+      {/* Error notification */}
+      {error && (
+        <ErrorNotification
+          error={new Error(error)}
```

Comment:
We convert to `Error` just to use the `error.message`? 
Maybe we should just use the error string itself